apps/menu-bar/src/components/DeviceItem.tsx

```diff
@@ -94,22 +95,35 @@ const DeviceItem = ({device, onPress, onPressLaunch, selected}: Props) => {
         )}
         {device.deviceType !== 'device' && device.state === 'Booted' && (
           <>
-            <Text color="success" style={styles.runIndicator}>
+            <Text color="success" style={styles.indicator}>
               ●
             </Text>
-            <Text color="secondary" style={styles.runIndicator}>
+            <Text color="secondary" style={styles.indicator}>
               {' '}
               Running
             </Text>
           </>
         )}
+        {device.deviceType !== 'device' &&
+          device.state === 'Shutdown' &&
+          isDeviceLaunching && (
+            <Text color="secondary" style={styles.indicator}>
+              Launching…
+            </Text>
+          )}
         {isHovered &&
           device.deviceType !== 'device' &&
-          device.state === 'Shutdown' && (
+          device.state === 'Shutdown' &&
+          !isDeviceLaunching && (
             <Button
+              disabled={isDeviceLaunching}
               color="primary"
-              onPress={onPressLaunch}
-              style={styles.button}>
+              onPress={async () => {
+                setDeviceLaunching(true);
+                await onPressLaunch();
+                setDeviceLaunching(false);
```

Comment:
Let's add a try-catch here just in case

```suggestion
                setDeviceLaunching(true);
                try {
                    await onPressLaunch(); 
                } catch (error) {
                     console.warn(error);
                } finally {
                    setDeviceLaunching(false);
                }
```